### PR TITLE
Feat/unify parameter

### PR DESCRIPTION
**Pull request by apenzk**
# Description of change

This PR is used to combine the parameter updates from issue #1507 

- Use the time.Duration type instead of specifying seconds in integers.
- Parameter names in `lowerCamelCase`
- `autopeering.seed` renamed to `node.seed`
- `network.externalAddress` renamed to `node.externalAddress`
- `port` in gossip and auto peering is not `bindAddress`

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## Change checklist

Add an `x` to the boxes that are relevant to your changes, and delete any items that are not.

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by Wollac — APPROVED** [review]

File: plugins/autopeering/local/local.go [hunk 1]
```diff
@@ -78,6 +70,10 @@ func configureLocal() *peer.Local {
 	// key, _ := peerDB.LocalPrivateKey()
 	// fmt.Printf("Seed: base58:%s\n", key.Seed().String())
 
+	// TODO: remove requirement for PeeringKey in hive.go
```
Comment (Wollac) [inline plugins/autopeering/local/local.go:73]: `NewLocal` requires a peering service to be present during construction. As now the identity is independent of the auto peering, this requirement should be removed in `hive.go`. However, due to version conflicts, dummy this dummy solution first seems easier.

File: deploy/ansible/initial-setup.yml [hunk 2]
```diff
@@ -12,7 +12,7 @@
     - role: goshimmer-node
       nodeType: entrynode
```
Comment (acha-bill) [inline deploy/ansible/initial-setup.yml:13]: should be camelCase.

File: deploy/ansible/deploy.yml [hunk 3]
```diff
@@ -12,7 +12,7 @@
     - role: goshimmer-node
```
Comment (acha-bill) [inline deploy/ansible/deploy.yml:12]: Roles too should be camelCase no?

File: deploy/ansible/roles/goshimmer-node/templates/docker-compose-analysis-server.yml.j2 [hunk 4]
```diff
@@ -4,7 +4,7 @@ version: "3"
 services:
   analysis_server:
```
Comment (acha-bill) [inline deploy/ansible/roles/goshimmer-node/templates/docker-compose-analysis-server.yml.j2:5]: should be camelCase

File: plugins/analysis/client/heartbeat.go [hunk 5]
```diff
@@ -7,9 +7,10 @@ import (
 	"github.com/iotaledger/hive.go/identity"
 	"github.com/mr-tron/base58"
 
+	"github.com/iotaledger/goshimmer/plugins/autopeering"
+
```
Comment (acha-bill) [inline plugins/analysis/client/heartbeat.go:11]: Format import

File: plugins/analysis/dashboard/frontend/package.json [hunk 6]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "goshimmer-analysis-dashboard",
+  "name": "goshimmer-analysisDashboard",
```
Comment (acha-bill) [inline plugins/analysis/dashboard/frontend/package.json:2]: why not camelCase everything?

File: plugins/messagelayer/consensus_plugin.go [hunk 7]
```diff
@@ -18,6 +18,8 @@ import (
 	"google.golang.org/grpc"
 	"google.golang.org/protobuf/proto"
 
+	"github.com/iotaledger/goshimmer/plugins/autopeering/discovery"
+
```
Comment (acha-bill) [inline plugins/messagelayer/consensus_plugin.go:22]: format import

File: plugins/metrics/plugin.go [hunk 8]
```diff
@@ -10,14 +10,15 @@ import (
 	"github.com/iotaledger/hive.go/node"
 	"github.com/iotaledger/hive.go/timeutil"
 
+	"github.com/iotaledger/goshimmer/plugins/autopeering"
+
```
Comment (acha-bill) [inline plugins/metrics/plugin.go:14]: format import

**Review by acha-bill — APPROVED** [review]
Nice :)

File: plugins/analysis/dashboard/frontend/package.json (hunk 6 above)
Comment (Wollac) [reply plugins/analysis/dashboard/frontend/package.json]: I think we should rather not touch the frontend code at all. Changed

File: config.default.json [hunk 9]
```diff
@@ -11,7 +11,7 @@
       "dev": false
     }
   },
-  "autopeering": {
+  "autoPeering": {
```
Comment (luca-moser) [inline config.default.json:14]: This should also include the change where the port is now a `bindAddress`

File: tools/docker-network/config.docker.json [hunk 10]
```diff
@@ -11,7 +11,7 @@
       "dev": false
     }
   },
-  "autopeering": {
+  "autoPeering": {
```
Comment (luca-moser) [inline tools/docker-network/config.docker.json:14]: Same thing here, `port` should be `bindAddress`

**Review by luca-moser — CHANGES_REQUESTED** [review]

**Review by luca-moser — APPROVED** [review]

**Pull request merged**